packages/devtools/lib/src/timeline/timeline_controller.dart

```diff
@@ -105,15 +118,117 @@ class TimelineController {
       timelineData.processTraceEvent(event, immediate: true);
     }
     // Make a final call to [maybeAddPendingEvents] so that we complete the
-    // processing for every frame in the import.
+    // processing for every frame in the snapshot.
     timelineData.maybeAddPendingEvents();
   }
 
-  void exitImportMode() {
+  void exitSnapshotMode() {
     // If the timeline controller had previously been started, restart it
-    // because [_timelineData] has changed since we entered import mode.
+    // because [_timelineData] has changed since we entered snapshot mode.
     if (hasStarted) {
       startTimeline();
     }
   }
 }
+
+class TimelineSnapshot {
```

Comment:
It would be ideal if the model for the snapshot was more consistent with the model in the regular mode. For example, the data model consists of all these things when viewing regular data as well. the main limitation when dealing with a snapshot is fewer extra interactions are possible.